src/Promitor.Scraper.Tests.Unit/Serialization/v1/Providers/ContainerRegistryDeserializerTests.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.ComponentModel;
+using Microsoft.Extensions.Logging;
+using Moq;
+using Promitor.Core.Scraping.Configuration.Serialization;
+using Promitor.Core.Scraping.Configuration.Serialization.v1.Model;
+using Promitor.Core.Scraping.Configuration.Serialization.v1.Model.ResourceTypes;
+using Promitor.Core.Scraping.Configuration.Serialization.v1.Providers;
+using Xunit;
+
+namespace Promitor.Scraper.Tests.Unit.Serialization.v1.Providers
+{
+    [Category("Unit")]
+    public class ContainerRegistryDeserializerTests : ResourceDeserializerTestBase
+    {
+        private readonly ContainerRegistryDeserializer _deserializer;
+
+        public ContainerRegistryDeserializerTests()
+        {
+            _deserializer = new ContainerRegistryDeserializer(new Mock<ILogger>().Object);
+        }
+
+        [Fact]
+        public void Deserialize_RegistryNameSupplied_SetsRegistryName()
+        {
+            DeserializerTestHelpers.AssertPropertySet<ContainerRegistryResourceV1, AzureResourceDefinitionV1, string>(
```

Comment:
I might be a bit oldskool but I prefer a more of a Arrange / Act / Assert approach as it's easier to read the tests imo.

Is it possible to:
1. Move input to const variables such as the input YAML
2. Do serialization
3. Assert on result?

Reply:
Obviously this one is just an example and applies a bit to all of these tests

Reply:
Yeah - if you take a look at all the tests I've added as a whole, you'll see that I do split them into Arrange / Act / Assert. The reason I didn't in this case is because the tests are identical for testing all the fields for a deserializer.

If you take a look at the implementation of the `DeserializerTestHelpers` class, you'll see that each test is split into Arrange, Act, Assert. I've only done it in this particular scenario because each test has a single call to the test method, so we're not in danger of not being able to understand what the test is doing, or of mixing the three test stages for example.

So I can rewrite all these tests if you want, but in this particular situation I think we'll just end up with a massive heap of boiler plate code and no benefit.

Reply:
That makes sense and is ok, maybe we should just tweak the naming a bit then?

i'm not really a fan of calling things "helper", but that's my personal preference.

How about `AzureResourceSerializationTests.DeserializeAndAssert` or `AzureResourceSerializationTests.DeserializeAndAssertProperty`? Does that make sense?

(Sorry, I always use long names but they tend to be explicit what they do)

Reply:
Yeah - I agree. I just hadn't come up with a decent name, and went with that to let me get on with implementing the changes. As per your other comment, I've renamed the class to `YamlAssert` so we now have `YamlAssert.PropertySet()`, etc.

Reply:
Thanks!